common/util/src/main/java/org/thingsboard/common/util/ThingsBoardScheduledThreadPoolExecutor.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright © 2016-2024 The Thingsboard Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.thingsboard.common.util;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+final class ThingsBoardScheduledThreadPoolExecutor extends ScheduledThreadPoolExecutor {
+
+    public ThingsBoardScheduledThreadPoolExecutor(int corePoolSize, ThreadFactory threadFactory) {
+        super(corePoolSize, threadFactory);
+    }
+
+    @Override
+    protected void afterExecute(Runnable r, Throwable t) {
+        super.afterExecute(r, t);
+        logExceptionsAfterExecute(r, t);
+    }
+
+    private static void logExceptionsAfterExecute(Runnable r, Throwable directThrowable) {
+        Throwable wrappedThrowable = extractThrowableWrappedInFuture(r);
```

Comment:
I see no point to extract all the time. We already log an error if the wrapped runnable has exception

Here we will get a Throwable if the Wrapper runnable `SafePeriodicRunnable` will get a Cancel or shutdown command. 

No go according performance as well.